linkedList/app/src/main/java/linkedlist/LinkedList.java

```diff
@@ -82,6 +88,48 @@ public void insertAfter(int searchValue, int newValue){
         curr.next = newNode;
     }
 
+    public int kthEnd(int k) throws IllegalArgumentException{
+
+        // check to see if tail node is there if not break out
+        if (this.tail == null){
+            throw new IllegalArgumentException("this list does not have a tail");
+        }
+
+        //create curr node for the tail
+        Node curr = this.tail;
+        //need int location counter
+        int locationCounter = 0;
+
+        // if it is there start array from tail and iterate backwards
+        while (curr != null){
+//             COMMENTED OUT COULD NOT PASS TEST
+//            if(locationCounter > k){
```

Comment:
This `if` statement will never be true inside this `while` loop. When `curr` reaches the head of the list, `locationCounter` will be the same length as the list. If `k` is longer than the length of the list, at this point, `locationCounter > k` will evaluate to `false`. On the next iteration, `curr` will be null, the `while` loop will break, and it will throw the exception on line 126.